task.ps1

```
@@ -6,7 +6,15 @@
 $vnetAddressPrefix = "10.0.0.0/16"
 $subnetAddressPrefix = "10.0.0.0/24"
 $sshKeyName = "linuxboxsshkey"
-$sshKeyPublicKey = Get-Content "~/.ssh/id_rsa.pub" 
+
+# Read SSH key
+$sshKeyPath = Join-Path $HOME ".ssh/id_ed25519.pub"
+if (-not (Test-Path $sshKeyPath)) {
+    Write-Error "SSH public key not found at $sshKeyPath. Please generate or specify a valid key before running this script."
+    exit 1
```

If the SSH key file is not found you exit immediately. Good to fail early, but consider offering a helpful hint or a parameter to supply an alternate key path instead of requiring manual edits. This will help testers who have different default key filenames (id_rsa.pub etc.).